tools/deploy-with-olm.sh

```diff
@@ -47,24 +58,42 @@ exit_on_err make docker-push IMG=${DEPLOYER_IMAGE}
 # Generate the olm bundle image
 blue "Generate and push the olm bundle image: ${BUNDLE_IMAGE}"
 exit_on_err make bundle IMG=${DEPLOYER_IMAGE}
+exit_on_err sed -i "s|sed_replace_addon_name|${ADDON_NAME}|" ${BUNDLE_FILE}
+exit_on_err sed -i "s|sed_replace_sop_endpoint|${SOP_ENDPOINT}|" ${BUNDLE_FILE}
 exit_on_err make bundle-build BUNDLE_IMG=${BUNDLE_IMAGE}
 exit_on_err make docker-push IMG=${BUNDLE_IMAGE}
 
 # In install olm if needed
 blue "Checking for existing olm"
-${K8S_CLIENT} get namespace olm >/dev/null 2>&1
-if [ $? -eq 0 ]; then
+if ${K8S_CLIENT} get namespace olm >/dev/null 2>&1
+then
     blue "olm found"
+    OLM_NAMESPACE="olm"
+elif ${K8S_CLIENT} get namespace openshift-operator-lifecycle-manager &> /dev/null
+then
+    blue "olm found"
+    OLM_NAMESPACE="openshift-operator-lifecycle-manager"
 else 
     blue "olm not found, installing olm in target cluster"
     exit_on_err ${OP_SDK} olm install     
 fi
 
 # Deploy the bundle
 blue "Deploy/Run the ${BUNDLE_NAME}:${BUNDLE_VERSION} in the cluster (at ${TARGET_NAMESPACE})"
-${K8S_CLIENT} get namespace olm >/dev/null 2>&1
-if [ $? -eq 0 ]; then
-    exit_on_err ${K8S_CLIENT} create namespace ${TARGET_NAMESPACE}
-fi
-exit_on_err ${K8S_CLIENT} apply -f ./tools/ocs-catalogsource.yaml
+
+exit_on_err ${K8S_CLIENT} create namespace ${NAMESPACE} --dry-run=client -o yaml | kubectl apply -f -
```

Comment:
${K8S_CLIENT} instead of kubectl as well?

Reply:
Yes, that's defined in line 19 and comes from the original script. The default value is kubectl.

Reply:
We use an env var so you could specify `oc` instead of `kubectl` when you run

Reply:
Oh, I used `kubectl` instead of `${K8S_CLIENT}` at the end of this line by accident. Good catch I'll change that.